### PR TITLE
[calcite-tree] Add break-word for long wrapping tree items

### LINKED ISSUE #605: calcite-tree category title overruns its bounds
An expandable heading in a calcite-tree overruns its bounds if it has long non-breaking word(s) (e.g., pseudo strings from the i18n team!).  Its children are constrained to the container.

In the images below, a calcite-tree is contained within a 150px-wide div represented by the green outline.
image
image

```html
<!doctype html>
<html>
<head>
  <link
    rel="stylesheet"
    type="text/css"
    href="https://unpkg.com/@esri/calcite-components@1.0.0-beta.26/dist/calcite/calcite.css"
  />
  <script
    type="module"
    src="https://unpkg.com/@esri/calcite-components@1.0.0-beta.26/dist/calcite/calcite.esm.js"
  ></script>
  <script
    nomodule=""
    src="https://unpkg.com/@esri/calcite-components@1.0.0-beta.26/dist/calcite/calcite.js"
  ></script>
</head>
<body>
  <div style="width:150px;border:1px solid green">
    <calcite-tree>
      <calcite-tree-item>
        <a href="#">This_is_a_very_long_nonbreaking_category_title</a>
        <calcite-tree slot="children">
          <calcite-tree-item>
            <a href="#">This_is_a_very_long_nonbreaking_category_entry</a>
          </calcite-tree-item>
        </calcite-tree>
      </calcite-tree-item>
    </calcite-tree>
  </div>
</body>
</html>
```

### PR DESCRIPTION
**Pull request by macandcheese**
Fixes #605 cc @MikeTschudi

I don't think this will cause unexpected changes for anyone?

- Add `word-break: break-word` to slotted children to prevent overflow of long, unbroken strings
- Updates changelog

| Before  | After |
| ------------- | ------------- |
| Screen Shot 2020-05-20 at 11 17 12 AM | Screen Shot 2020-05-20 at 11 17 24 AM |

### CODE REVIEW

File: src/components/calcite-tree-item/calcite-tree-item.scss [hunk 1]
```diff
@@ -87,6 +87,7 @@
   text-decoration: none !important;
   max-width: 100%;
   word-wrap: break-word;
+  word-break: break-word;
```
Comment (paulcpederson) [inline src/components/calcite-tree-item/calcite-tree-item.scss:90]: I think we may actually want `overflow-wrap: break-word;` here, from [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-wrap):

> Note: In contrast to word-break, overflow-wrap will only create a break if an entire word cannot be placed on its own line without overflowing.

They've actually marked `word-break: break-word` as deprecated.

Comment (macandcheese) [reply src/components/calcite-tree-item/calcite-tree-item.scss:90]: Hmmm... I wasn't able to get `overflow-wrap` to work on any combination of wrapping elements 🤔

Comment (paulcpederson) [reply src/components/calcite-tree-item/calcite-tree-item.scss:90]: one second let me take a gander here

Comment (paulcpederson) [reply src/components/calcite-tree-item/calcite-tree-item.scss:90]: weird I think it only works if you also give it `display: block` or `display: inline-block`

**Comment by paulcpederson** [comment]
Nice.

**Comment by macandcheese** [comment]
Updated and added some IE11 specific width things 🔄

**Pull request merged**